File: Matrix/0036.Valid Sudoku/Solution.java
class Solution {
    public boolean isValidSudoku(char[][] board) {
        int[][] row = new int[9][10];
        int[][] col = new int[9][10];
        int[][][] box = new int[3][3][10];
        for (int i = 0; i < 9; i++) {
            for (int j = 0; j < 9; j++) {
                if ('.' == board[i][j]) {
                    continue;
                }
                int v = board[i][j] - '0';
                if (row[i][v]++ > 0) {
                    return false;
                }
                if (col[j][v]++ > 0) {
                    return false;
                }
                if (box[i / 3][j / 3][v]++ > 0) {
                    return false;
                }
            }
        }
        return true;
    }
}

File: Matrix/0036.Valid Sudoku/Solution.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        row = [[0] * 10 for _ in range(9)]
        col = [[0] * 10 for _ in range(9)]
        box = [[[0] * 10 for _ in range(3)] for _ in range(3)]
        for x in range(9):
            for y in range(9):
                if board[x][y] == '.':
                    continue
                v = int(board[x][y])
                if row[x][v] != 0:
                    return False
                else:
                    row[x][v] += 1
                if col[y][v] != 0:
                    return False
                else:
                    col[y][v] += 1
                if box[x // 3][y // 3][v] != 0:
                    return False
                else:
                    box[x // 3][y // 3][v] += 1
        return True